SCClassLibrary/Common/Control/Server.sc

```diff
@@ -375,18 +377,27 @@ Server {
 	}
 
 	initTree {
-		this.newNodeAllocators;
 		this.sendDefaultGroups;
 		tree.value(this);
-		ServerTree.run(this);
+		Routine { ServerTree.run(this) }.play(AppClock);
 	}
 
 	/* id allocators */
 
-	// private, called from server notify response with next free clientID
+	isFullyBooted { ^this.serverRunning and: { this.notified } }
```

Comment:
Actually, thinking about the naming: To me it seems that `server.serverRunning` should return only ever `true` when it is notified.

so I'd suggest:

```
var <>hasBooted;

serverRunning { ^this.hasBooted and: { this.notified } }
```

This might mean more changes internally. But semantically it seems correct. E.g. when you start a `NodeProxy` it should warn also when the server hasn't been notified yet. Otherwise we'll run into allocator troubles.

Reply:
Or not?

Reply:
good point - yes can try that. a bit hairy to get to work properly, but should leave booting cleaner for the next reworking.

Reply:
will you still add it in this PR?